pkg/build/controller/strategy/docker.go

```diff
@@ -171,8 +225,19 @@ func (bs *DockerBuildStrategy) CreateBuildPod(build *buildapi.Build) (*v1.Pod, e
 	}
 
 	setOwnerReference(pod, build)
-	setupDockerSocket(pod)
-	setupCrioSocket(pod)
+	setupSockets := true
+	if s := build.Spec.Strategy.DockerStrategy; s != nil {
+		if policy := s.ImageOptimizationPolicy; policy != nil {
+			switch *policy {
+			case buildapi.ImageOptimizationDaemonless, buildapi.ImageOptimizationDaemonlessWithLayers, buildapi.ImageOptimizationDaemonlessSquashed:
+				setupSockets = false
+			}
+		}
+	}
+	if setupSockets {
+		setupDockerSocket(pod)
+		setupCrioSocket(pod)
```

Comment:
buildah doesn't need this(right?) so ultimately i'd expect this to go away.  ie if we're running on a docker host, we'd just use the docker socket and if we're running on a crio host, we'd use buildah and not need any container sockets.